### ISSUE TITLE
count():Parameter.....onlyoffice/lib/appconfig.php#477 error in logs

### ISSUE THREAD

**Issue by osscombat** [ONLYOFFICE/onlyoffice-nextcloud#62]
This is a re-opened [issue #43](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/issues/43) cause the fix is not yet confirmed in GA releases.

OnlyOffice-nextcloud v.2.1.2
onlyoffice-documentserver v.5.2.4
NC 14.0.6

Recently after the upgrades to these versions I noticed that upon every user logon to NC I get this message in the NC log:

count(): Parameter must be an array or an object that implements Countable at /var/www/html/nextcloud/apps/onlyoffice/lib/appconfig.php#477

and sometime

implode(): Invalid arguments passed at /var/www/html/nextcloud/apps/onlyoffice/templates/settings.php#66

I see no other problems, everything works fine though.

**Comment by Cliffhou** [comment]
Same issue here

**Comment by Cliffhou** [comment]
count(): Parameter must be an array or an object that implements Countable at /var/www/nextcloud/apps/onlyoffice/lib/appconfig.php#477

**Comment by Cliffhou** [comment]
And occasionally this error show up:
Undefined offset: 2 at /var/www/nextcloud/apps/onlyoffice/controller/editorcontroller.php#434

**Comment by raceface2nd** [comment]
I have the same error showing up

`Undefined offset: 2 at /var/www/nextcloud/apps/onlyoffice/controller/editorcontroller.php#434`

... and recognized that if files are edited in the browser using onlyoffice nextcloud app, nextcloud is not recognizing that the files are edited unless the files cache is cleared. Some of my users complained that NC showed that some files has not been changed since days even they changed the file in the onlyoffice app

Nextcloud 14.0.6 with onlyoffice 2.1.2 and only office server 5.2.4.94 (docker container on univention)

**Comment by alexanderonlyoffice** [comment]
Hello @osscombat, @Cliffhou, @raceface2nd, this message doesn't affect document editing in Nextcloud, it just appears in logs. The issue will be fixed in the next version of ONLYOFFICE connector app.

**Comment by sesipod** [comment]
I have just reinstalled my Nextcloud server and I am getting the following errors:

<img width="1204" alt="Screen Shot 2019-03-11 at 8 43 11 AM" src="https://user-images.githubusercontent.com/1533744/54128284-b9761a80-43d9-11e9-9f30-a9140e398fce.png">

any time I try and edit a document I get the following error: 

<img width="931" alt="Screen Shot 2019-03-11 at 8 44 00 AM" src="https://user-images.githubusercontent.com/1533744/54128328-d7437f80-43d9-11e9-868e-6a1de87d0ce6.png">

**Comment by sesipod** [comment]
Fixed my issue by completely wiping out the Docker Server and re-downloading the latest version. 

```
sudo docker run --memory=2.5g -i -t -d -p 80:80 \
    -v /app/onlyoffice/DocumentServer/logs:/var/log/onlyoffice  \
    -v /app/onlyoffice/DocumentServer/data:/var/www/onlyoffice/Data  \
    -v /app/onlyoffice/DocumentServer/lib:/var/lib/onlyoffice \
    -v /app/onlyoffice/DocumentServer/db:/var/lib/postgresql  onlyoffice/documentserver
```

**Comment by bi1012037** [comment]
PHP 7.3
Apache/2.4.18
I change /var/www/owncloud/apps/onlyoffice/lib/appconfig.php file.
        $groups = $this->GetLimitGroups();
        // no group set -> all users are allowed
        if (count($groups) === 0) {
            return true;
        }
to
        $groups = $this->GetLimitGroups();
        // no group set -> all users are allowed
        if (@count($groups) === 0) {
            return true;
        }
fix #477,but find another questions (#434 and #66).
When user use onlyoffice to open .docx file,the bugs will appear. 
![error1](https://user-images.githubusercontent.com/20457938/54509990-9376ea00-4986-11e9-89f2-3a486459f8b4.PNG)
![error2](https://user-images.githubusercontent.com/20457938/54509993-9671da80-4986-11e9-84b7-c18201240168.PNG)

**Comment by ChristophWurst** [comment]
I think this was fixed with https://github.com/ONLYOFFICE/onlyoffice-nextcloud/commit/751fbcf732e4c37168ae9328983093e2c6668267 https://github.com/ONLYOFFICE/onlyoffice-nextcloud/pull/71
